pkg/sql/plan/function/builtin/multi/serial.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2021 - 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multi
+
+import (
+	"github.com/matrixorigin/matrixone/pkg/container/nulls"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/errno"
+	"github.com/matrixorigin/matrixone/pkg/sql/errors"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+)
+
+func Serial(vectors []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	length := vector.Length(vectors[0])
+	for _, v := range vectors {
+		for i := 0; i < length; i++ {
+			if nulls.Contains(v.Nsp, uint64(i)) {
+				return nil, errors.New(errno.DataException, "serial function don't support null value")
+			}
+		}
+		if v.IsScalar() {
+			return nil, errors.New(errno.DataException, "serial function don't support null value")
```

Comment:
typo? null or scalar?

Reply:
serial is used to build composite primary key, we don't allow null. In the build composite primary key, scalar vector don't appear, so I don't allowed

Reply:
serial function don't support constant value
is suitable for the case v.IsScalar()

Reply:
> serial function don't support constant value is suitable for the case v.IsScalar()
ok, i change it